refnx/reflect/_mdsimulation.py

```diff
@@ -0,0 +1,307 @@
+            Only required if the radiation is 'xray'. This is the energy of the
+            probing X-ray radiation, in units of kilo electron Volts
+        lgtfile: str, optional
+            The path and name of the .lgt file (if present), which contains the
+            scattering lengths of each of the atom type in the pdbfile.
+            Currently the .lgt file style that is supported is a 3 column space
+            separated txt file where the columns are atom_type,
+            real_scattering_length, and imaginary_scattering_length
+            respectively. If a lgtfile is not used the scattering lengths for
+            the system will be determined based on the element type that is
+            defined in the final column of the pdb file.
+        atom_types: str, array_like
+            An array of length N, where N is the number of atom types in the
+            simulation trajectory, which defines the atom types to assign
+            scattering lengths.
+        scattering_lengths: float, array_like
+            An array of shape [N, 2], where N is the number of atom types in
+            the simulation trajectory, which defines the scattering lengths for
+            the given atom types.
+        """
+        self.scatlens = {}
+        if atom_types and scattering_lengths:
+            if len(atom_types) == len(scattering_lengths):
+                for i in range(0, len(atom_types)):
+                    self.scatlens[atom_types[i]] = scattering_lengths[i]
+            else:
+                raise ValueError('The lengths of the atom types must be the '
+                                 'same as the lengths of the pairs of '
+                                 'scattering lengths.')
+        elif lgtfile:
+            self.lgtfile = lgtfile
+            self.read_lgt()
+        else:
+            if radiation == 'neutron':
+                self.neutron = True
+            else:
+                if not xray_energy:
+                    raise ValueError('If the probing radiation is the X-ray'
+                                     ' it is necessary to define an '
+                                     'xray_energy (in keV).')
+                else:
+                    self.neutron = False
+                    self.xray_energy = xray_energy
+            self.get_lgts_from_pt()
+        if self.verbose:
+            print('Scattering lengths found.')
+
+    def run(self):
+        """
+        Calculate the scattering profile from the simulation trajectory.
+        """
+        self._get_sld_profile()
+        if self.verbose:
+            print('SLD profile determined.')
+
+    def read_pdb(self):
+        """Parse pdb file.
+
+        Parses the pdbfile.
+        """
+        parser = PDBParser()
+        self.structure = parser.get_structure('model', self.pdbfile)
+        file = open(self.pdbfile, 'r')
+        for line in file:
+            if line.startswith('CRYST1') and not np.any(self.dimensions):
```

Comment:
Is this file parsing bit robust - is there any variation in PDF file that causes this to fall over?

Reply:
The amount of variation in the pdb file format is sickening, to be honest (a headache for many people). That said, this is probably the most robust way possible, as often pdb files are output with things in incorrect columns, so the column approach that really should be used will fall down.